pkg/cmd/roachtest/testselector/snowflake_query.sql

```diff
@@ -36,10 +41,15 @@ select
          failure_count > 0 or -- the test has failed at least once in the past "forPastDays" days
          first_run > dateadd(DAY, ?, current_date()) or -- recently added test - test has not run for more than "firstRunOn" days
          last_run < dateadd(DAY, ?, current_date()) or -- the test has not been run for last "lastRunOn" days
-         last_run is null -- the test is always ignored till now or have never been run
-         then 'yes' else 'no' end as selected,
+         -- as stated above, latest unsuccessful test should be selected. this will not consider any test
+         -- that has failed with non-infra error as the failure_count will not be zero.
+         recent_status != 'SUCCESS'
```

Comment:
I'm not sure I understand what this predicate is supposed to capture. What test runs is this supposed to include?

Reply:
basically, if a test is successful recently, but have failed in the past 30 days should be selected. But, as per this condition, those tests will get ignored. However, the `failure_count > 0` or condition will handle this.